File: app/src/main/java/com/example/wpossbank/DepositsActivity.java
package com.example.wpossbank;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;

import com.example.wpossbank.fragments.Dialogs;
import com.example.wpossbank.modelos.Admin;
import com.example.wpossbank.modelos.MakeMessages;
import com.example.wpossbank.modelos.User;
import com.example.wpossbank.modelos.Validate;

import eightbitlab.com.blurview.BlurView;
import eightbitlab.com.blurview.RenderScriptBlur;

public class DepositsActivity extends AppCompatActivity {
    Context context;
    Validate validate;
    MakeMessages messages;

    Admin admin;
    User user;

    BlurView blurView;

    EditText ccInput, ccDepositInput, depositInput;
    Button confirmButton;
    ImageView backArrow;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_deposit);
        context = this;
        validate = new Validate(context);
        messages = new MakeMessages();

        admin = new Admin();
        user = new User(context);

        blurView = findViewById(R.id.blurView);
        blurBackground();

        ccInput = findViewById(R.id.ccInput);
        ccDepositInput = findViewById(R.id.ccDepositInput);
        depositInput = findViewById(R.id.depositInput);

        confirmButton = findViewById(R.id.confirmButton);
        backArrow = findViewById(R.id.backArrow);

        user.loadData(user);//Carga la información del usuario desde la base de datos

        confirmButton.setOnClickListener(confirmDeposit -> {
            //Deckara y verifica si los campos tienen la información correcta
            boolean ccValidate = validate.isNumber(ccInput) && validate.matchUserData(ccInput, "cc"),
                    ccDepositValidate = validate.isNumber(ccDepositInput) && validate.isInRange(ccDepositInput,10,13),
                    depositValidate = validate.isInRange(depositInput,2,7);

            if (ccValidate && ccDepositValidate && depositValidate){
                int depositValue = Integer.parseInt(depositInput.getText().toString());

                admin.setBalance(admin.getCost()/2);
                new Dialogs.ConfirmUserAddBalance(context ,admin,
                            messages.adminDeposit(context, depositInput), "deposit",
                        user.getCc(), depositValue).show(getSupportFragmentManager(),"CONFIRM");
            }
        });

        backArrow.setOnClickListener( goBack -> finish());
    }

    private void blurBackground(){
        float radius = 20f;

        View decorView = getWindow().getDecorView();
        ViewGroup rootView = decorView.findViewById(android.R.id.content);
        Drawable windowBackground = decorView.getBackground();

        blurView.setupWith(rootView)
                .setFrameClearDrawable(windowBackground)
                .setBlurAlgorithm(new RenderScriptBlur(this))
                .setBlurRadius(radius)
                .setBlurAutoUpdate(true)
                .setHasFixedTransformationMatrix(true);
    }
}

File: app/src/main/java/com/example/wpossbank/activities/DepositsActivity.kt
package com.example.wpossbank.activities

import android.content.Context
import android.os.Bundle
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText
import android.widget.ImageView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.wpossbank.R
import com.example.wpossbank.fragments.Dialogs.ConfirmUserUpdateBalance
import com.example.wpossbank.models.*
import eightbitlab.com.blurview.BlurView
import eightbitlab.com.blurview.RenderScriptBlur

class DepositsActivity : AppCompatActivity() {
    lateinit var context: Context
    lateinit var validate: Validate
    lateinit var messages: MakeMessages
    lateinit var admin: Admin
    lateinit var user: User
    lateinit var blurView: BlurView
    lateinit var ccInput: EditText
    private lateinit var ccDepositInput: EditText
    private lateinit var depositInput: EditText
    lateinit var confirmButton: Button
    lateinit var backArrow: ImageView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_deposit)
        context = this

        validate = Validate(context)
        messages = MakeMessages()
        admin = Admin()
        user = User()

        blurView = findViewById(R.id.blurView)
        blurBackground()

        ccInput = findViewById(R.id.ccInput)
        ccDepositInput = findViewById(R.id.ccDepositInput)
        depositInput = findViewById(R.id.depositInput)
        confirmButton = findViewById(R.id.confirmButton)
        backArrow = findViewById(R.id.backArrow)

        user.loadData(context) //Carga la información del usuario desde la base de datos

        confirmButton.setOnClickListener {
            //Deckara y verifica si los campos tienen la información correcta
            val ccValidate = validate.isNumber(ccInput) && validate.matchUserData(ccInput, "cc")
            val ccDepositValidate =
                validate.isNumber(ccDepositInput) && validate.isInRange(ccDepositInput, 10, 13)
            val depositValidate = validate.isInRange(depositInput, 2, 7)
            if (ccValidate && ccDepositValidate && depositValidate) {
                val depositValue = depositInput.text.toString().toInt()
                admin.balance = admin.cost / 2

                val logEntry = LogEntry()
                logEntry.type = "deposit"
                logEntry.source = user.cc
                logEntry.amount = depositValue

                ConfirmUserUpdateBalance(context, logEntry, admin, messages.deposit(context, depositInput))
                    .show(supportFragmentManager, "CONFIRM")
            } else {
                // Avisa al usuario si hay un campo con valores invalidos
                Toast.makeText(context, R.string.error_invalid_input, Toast.LENGTH_LONG).show()
            }
        }
        backArrow.setOnClickListener { finish() }
    }

    private fun blurBackground() {
        val radius = 20f
        val decorView = window.decorView
        val rootView = decorView.findViewById<ViewGroup>(android.R.id.content)
        val windowBackground = decorView.background
        blurView.setupWith(rootView)
            .setFrameClearDrawable(windowBackground)
            .setBlurAlgorithm(RenderScriptBlur(this))
            .setBlurRadius(radius)
            .setBlurAutoUpdate(true)
            .setHasFixedTransformationMatrix(true)
    }
}